profondo/cnn.py

```diff
@@ -86,6 +95,96 @@ def build_model(x_train, y_train):
     return model
 
 
+def build_pre_trained(network_name, x_train, y_train, x_val, y_val):
+    supported_models = {
+        'vgg16': {
+            'class': kapps.VGG16,
+            'params': {
+                'weights': 'imagenet',
+                'include_top': False,
+                'input_shape': x_train.shape[1:]
+            }
+        }
+    }
+    
+    if not network_name in supported_models:
+        raise Exception("Unsupported pre-trained model {}".format(network_name))
+
+    # Load the convolutional part from a pre-trained model.
+    model_params = supported_models[network_name]['params']
+    model = supported_models[network_name]['class'](**model_params)
+
+    # Build a classifier model to put on top of the convolutional model.
+    def generate_top_model(input_shape, output_shape):
+        m = Sequential()
+        m.add(Flatten(input_shape=input_shape))
+        m.add(Dense(4096, activation='relu'))
+        m.add(Dropout(0.5))
+        m.add(Dense(1024, activation='relu'))
+        m.add(Dropout(0.25))
+        m.add(Dense(output_shape, activation='sigmoid'))
+        return m
+
+    if not exists(TOP_MODEL_WEIGHTS_FILE):
+        # Build the "bottleneck-features" to feed into the top layers, to init
+        # the output layers with reasonable features: we basically record the
+        # output of the last convolutional block with our data and feed that
+        # into the new top block.
+        # Note: our images are already scaled in the [0, 1] range, we don't need
+        # the "rescale" part as required by the pre-trained networks. We
+        # don't shuffle the samples to be able to pick their labels later on.
+        init_batch = 100
+        num_train_batches = len(x_train) // init_batch
+        bottleneck_train = model.predict(x_train, num_train_batches, verbose=True)
```

Comment:
I don't think `num_train_batches` makes sense how it's defined and used when passed to the `batch_size` argument of `model.predict()`. Can we just pass `args.batch_size`?